bigtable/tests/mutations_integration_test.cc

```diff
@@ -0,0 +1,555 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/testing/table_integration_test.h"
+
+namespace {
+
+namespace admin_proto = ::google::bigtable::admin::v2;
+
+class MutationIntegrationTest : public bigtable::testing::TableIntegrationTest {
+ protected:
+  std::string column_family1 = "fam1";
+  std::string column_family2 = "fam2";
+  std::string column_family3 = "fam3";
+
+  bigtable::TableConfig table_config = bigtable::TableConfig(
+      {{column_family1, bigtable::GcRule::MaxNumVersions(10)},
+       {column_family2, bigtable::GcRule::MaxNumVersions(10)},
+       {column_family3, bigtable::GcRule::MaxNumVersions(10)}},
+      {});
+
+  // A helper function to create a list of cells.
+  void CreateCellsIgnoringTimestamp(bigtable::Table& table,
+                                    std::vector<bigtable::Cell> const& cells) {
+    std::map<std::string, bigtable::SingleRowMutation> mutations;
+    for (auto const& cell : cells) {
+      std::string key = cell.row_key();
+      auto inserted = mutations.emplace(key, bigtable::SingleRowMutation(key));
+      inserted.first->second.emplace_back(bigtable::SetCell(
+          cell.family_name(), cell.column_qualifier(), cell.value()));
+    }
+
+    bigtable::BulkMutation bulk;
+    for (auto& kv : mutations) {
+      bulk.emplace_back(std::move(kv.second));
+    }
+    table.BulkApply(std::move(bulk));
+  }
+
+  /**
+   * Delete the records from table for the specified family and column
+   * identifier.
+   */
+  void DeleteFromColumn(bigtable::Table& table, std::string row_key,
+                        std::string column_family, std::string column_id) {
+    bigtable::SingleRowMutation mutation(row_key);
+    mutation.emplace_back(bigtable::DeleteFromColumn(column_family, column_id));
+
+    bigtable::BulkMutation bulk;
+    bulk.emplace_back(std::move(mutation));
+    table.BulkApply(std::move(bulk));
+  }
+
+  /**
+   * Delete the records from table for the specified family and column with
+   * timestamp starting from time_begin.
+   * time_begin: This parameter is inclusive, i.e. timestamp >= time_begin
+   */
+  void DeleteFromColumnStartingFrom(bigtable::Table& table, std::string row_key,
+                                    std::string column_family,
+                                    std::string column_id,
+                                    std::int64_t time_begin) {
+    bigtable::SingleRowMutation mutation(row_key);
+    mutation.emplace_back(bigtable::DeleteFromColumnStartingFrom(
+        column_family, column_id, time_begin));
+
+    bigtable::BulkMutation bulk;
+    bulk.emplace_back(std::move(mutation));
+    table.BulkApply(std::move(bulk));
+  }
+
+  /**
+   * Delete the records from table for the specified family and column with
+   * timestamp ending at time_end.
+   * time_end  : This parameter is not inclusive. i.e. timestamp < time_end
```

Comment:
Document arguments with `@param` please.

Reply:
Is it ok to  have the @params, only for the functions where I have added additional explanations
about the parameters. e.g. where time_begin and time_end parameters.

Reply:
Yes, if the parameters are all obvious there is no need to document them.  If you do need to document them please use the Doxygen directives though.  Unfortunately if you do need to document one parameter you need to document all of them.